src/layer.js

```diff
@@ -234,6 +234,13 @@ export class MapLayer extends HTMLElement {
       }
     }, 0);
   }
+  getOuterHTML() {
+    let fullLayer = this;
+    if (fullLayer.hasAttribute('src')) {
+      fullLayer.setAttribute('src', this._layer.getHref());
+    }
+    return fullLayer.outerHTML;
+  }
```

Comment:
`fullLayer` is just a reference to `this` (which is the `layer-` element in this case) so all of the changes you make to `fullLayer`, the same changes are also applied to `this`. This is why the layer element's src attribute gets updated when you copy that layer. 

You should not manipulate elements to get the desired results but instead can simply just manipulate a string, as that is what this method should return.